shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -1235,6 +1235,13 @@ public void onUserLeaveHint() {
     }
   }
 
+  @ActivityCallThrough
```

Comment:
Rather than requiring an explicit activity call through, you could use an [OnWindowFocusChangeListener](https://developer.android.com/reference/android/view/ViewTreeObserver.OnWindowFocusChangeListener) registered in `onViewCreated` and unregistered in `onDestroyView`. Something like

```java
private final OnWindowFocusChangeListener onWindowFocusChangeListener = 
    new OnWindowFocusChangeListener() {
      @Override
      public void onWindowFocusChanged(boolean hasFocus) {
        if (stillAttachedForEvent("onWindowFocusChanged")) {
          delegate.onWindowFocusChanged(hasFocus);
        }
      }
    }

@Override
public void onViewCreated(View view, Bundle savedInstanceState) {
  super.onViewCreated(view, savedInstanceState);
  view.getViewTreeObserver().addOnWindowFocusChangeListener(onWindowFocusChangeListener);
}

@Override
public void onDestroyView() {
  super.onDestroyView();
  requireView().getViewTreeObserver()
      .removeOnWindowFocusChangeListener(onWindowFocusChangeListener);
}
```
This assumes a minimum SDK version of 18+, so if Flutter supports older than that this may not be an option. And if you don't go this pattern, you need to update `FlutterFragmentActivity` to do the required call through.

Reply:
Flutter doesn't support 18.0 or below, so I think we're fine.

https://docs.flutter.dev/reference/supported-platforms

Reply:
> And if you don't go this pattern, you need to update FlutterFragmentActivity to do the required call through.

I already did that in this PR. 

What's the advantage of doing it the way you propose?

Reply:
The advantage is you don't need the activity call through. The logic is entirely self-contained to the fragment. Activity call through APIs are error-prone and difficult to work with, because a user could very easily forget to invoke them, and there isn't always an easy way to call through from the activity (such as a FlutterFragment nested within X other fragments). In a perfect world, there wouldn't be any methods in FlutterFragment requiring an activity call through.

Also, you need to update `shell/platform/android/io/flutter/embedding/android/FlutterFragmentActivity.java`. There's two of them, and the other one that you updated is (as I understand) deprecated.

Reply:
This a much better way to avoid the issue I was trying to prevent with a linter. Host apps are less likley to mess our their own state by failing to pass calls through. 

Context: https://github.com/flutter/engine/pull/41094#issuecomment-1518314588